### PR TITLE
Subs table membership user cols

### PR DESCRIPTION
**Pull request by dparker1005**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Moved the following columns from the `pmpro_memberships_users` table into `pmpro_subscriptions`:
- initial_payment
- billing_amount
- cycle_number
- cycle_period
- billing_limit
- trial_amount
- trial_limit

### CODE REVIEW

File: classes/class-pmpro-subscription.php [hunk 1]
```diff
@@ -146,9 +216,13 @@ public function __construct( $subscription = null ) {
 
 		if ( ! empty( $subscription_data ) ) {
 			foreach ( $subscription_data as $arg => $value ) {
-				// Enforce integers for properties.
+				// Enforce integers/floats for properties.
```
Comment (sc0ttkclark) [inline classes/class-pmpro-subscription.php:219]: @dparker1005 I just added this for you since we're introducing floats

Comment (dparker1005) [reply classes/class-pmpro-subscription.php:219]: Awesome, great catch. Thanks!

File: includes/upgradecheck.php [hunk 2]
```diff
@@ -398,13 +398,6 @@ function pmpro_db_delta()
 		   `user_id` bigint(20) unsigned NOT NULL,
 		   `membership_id` int(11) unsigned NOT NULL,
 		   `code_id` bigint(20) unsigned NOT NULL,
-		   `initial_payment` decimal(18,8) NOT NULL,
```
Comment (sc0ttkclark) [inline includes/upgradecheck.php:401]: Let's leave this section here until we handle migration work, wouldn't want to run it and break your tables when switching between branches

Comment (dparker1005) [reply includes/upgradecheck.php]: Just added this back

File: classes/class-pmpro-subscription.php [hunk 3]
```diff
@@ -217,7 +217,7 @@ public function __construct( $subscription = null ) {
 		if ( ! empty( $subscription_data ) ) {
 			foreach ( $subscription_data as $arg => $value ) {
 				// Enforce integers/floats for properties.
-				if ( isset( $this->{$arg} ) && is_int( $this->{$arg} ) ) {
+				if ( isset( $this->{$arg} ) && ( is_int( $this->{$arg} ) || is_float( $this->{$arg} ) ) ) {
```
Comment (sc0ttkclark) [inline classes/class-pmpro-subscription.php:220]:
```suggestion
				if ( isset( $this->{$arg} ) ) {
```

File: tests/_support/Factories/PMPro_SubscriptionFactory.php [hunk 4]
```diff
@@ -59,6 +66,13 @@ public function __construct( $factory = null, $default_generation_definitions =
 			'startdate'                   => gmdate( 'Y-m-d H:i:s' ),
 			'enddate'                     => '',
 			'next_payment_date'           => gmdate( 'Y-m-d H:i:s', strtotime( '+1 year' ) ),
+			'initial_payment'             => 0,
+			'billing_amount'              => 0,
+			'cycle_number'                => 0,
+			'cycle_period'                => 'Month',
+			'billing_limit'               => 0,
+			'trial_amount'                => 0,
+			'trial_limit'                => 0,
```
Comment (sc0ttkclark) [inline tests/_support/Factories/PMPro_SubscriptionFactory.php:75]:
```suggestion
			'trial_limit'                 => 0,
```

File: tests/wpunit/PMPro_SubscriptionTest.php [hunk 5]
```diff
@@ -50,24 +50,54 @@ public function test___construct_with_id() {
...
 		$subscription_data = [
-			'id'                  => '1234',
-			'user_id'             => '2345',
-			'membership_level_id' => '3456',
-			'status'              => 'active',
+			'id'                          => '1234',
+			'user_id'                     => '2345',
+			'membership_level_id'         => '3456',
+			'status'                      => 'active',
+			'gateway'                     => 'stripe',
```
Comment (sc0ttkclark) [inline tests/wpunit/PMPro_SubscriptionTest.php:57]: Let's keep this test simplified and not focused on all of the args at the moment. The test__call method will cover things more thoroughly and this just ends up covering the same things as that at when all the extra args are used/checked.

File: tests/wpunit/PMPro_SubscriptionTest.php [hunk 6]
```diff
@@ -76,24 +106,54 @@ public function test___construct_with_array() {
...
 		// Basic test data, just enough to test with.
 		$subscription_data = (object) [
-			'id'                  => '1234',
-			'user_id'             => '2345',
-			'membership_level_id' => '3456',
-			'status'              => 'active',
+			'id'                          => '1234',
+			'user_id'                     => '2345',
+			'membership_level_id'         => '3456',
+			'gateway'                     => 'stripe',
```
Comment (sc0ttkclark) [inline tests/wpunit/PMPro_SubscriptionTest.php:112]: Let's keep this test simplified and not focused on all of the args at the moment. The test__call method will cover things more thoroughly and this just ends up covering the same things as that at when all the extra args are used/checked.

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**